analytics.js

```diff
@@ -669,6 +678,66 @@ class Analytics {
     );
   }
 
+  IsEventBlackListed(eventName, intgName) {
+    if (typeof eventName === "string" && eventName) {
+      const sdkIntgName = commonNames[intgName];
+      const intg = this.clientIntegrations.find(
+        (intg) => intg.name === sdkIntgName
+      );
+
+      const { blacklistedEvents, whitelistedEvents, eventFilteringOption } =
+        intg.config;
+
+      if (eventFilteringOption) {
+        switch (eventFilteringOption) {
+          // disabled filtering
+          case "disable":
+            return false;
+          // Blacklist is choosen for filtering events
+          case "blacklistedEvents":
+            const isValidBlackList =
+              blacklistedEvents &&
+              Array.isArray(blacklistedEvents) &&
+              blacklistedEvents.every((x) => x.eventName !== "");
+
+            if (isValidBlackList) {
+              return blacklistedEvents.find(
+                (eventObj) =>
+                  eventObj.eventName.trim().toUpperCase() ===
+                  eventName.trim().toUpperCase()
+              ) === undefined
+                ? false
+                : true;
+            } else {
+              return false;
+            }
+          // Whitelist is choosen for filtering events
+          case "whitelistedEvents":
+            const isValidWhiteList =
+              whitelistedEvents &&
+              Array.isArray(whitelistedEvents) &&
+              whitelistedEvents.some((x) => x.eventName !== "");
+            if (isValidWhiteList) {
+              return whitelistedEvents.find(
+                (eventObj) =>
+                  eventObj.eventName.trim().toUpperCase() ===
+                  eventName.trim().toUpperCase()
+              ) === undefined
+                ? true
+                : false;
+            } else {
+              return true;
+            }
+          default:
+            return false;
+        }
+      } else {
+        return false;
+      }
+    }
+    return false;
+  }
```

Comment:
Perform validation in the beginning and return from there itself. Makes this logic less nested and readable.

Reply:
done